components/SubmissionForm.vue

```diff
@@ -168,4 +188,46 @@ const validateFields = (e: Event) => {
     }
     submissionStore.submit()
 }
+
+const submissionInputFields = [
+    { field: 'location', validate: validations.validateGoogleMapsUrlInput, validationReference: isValidGoogleMapsUrl },
+    { field: 'lastName', validate: validations.validateUserSubmittedLastName, validationReference: isValidLastName },
+    { field: 'firstName', validate: validations.validateUserSubmittedFirstName, validationReference: isValidFirstName },
+    { field: 'selectLanguage1', validate: validations.validateFirstSpokenLanguage, validationReference: isValidPrimarySpokenLanguage },
+    { field: 'selectLanguage2', validate: validations.validateSecondSpokenLanguage, validationReference: isValidSecondarySpokenLanguage }
+]
+
+const initialValidationCheck =
+async (inputValue: string, inputValidationCheck: (inputValue: string) => boolean, field: ValidationReference) => {
+    validationCheckedPreviously.value = true
+    await nextTick()
+    const validationReference = useCorrectValidation(field)
```

Comment:
after figuring out the clever goodness you're writing here, I think we can simplify this logic in one of two ways. 

### Approach 1
Why: A bit simpler, but still keeps your genericized approach

How: you could pass in the `isValid__` variable directly, then you don't need the `useCorrectValidation` method. 

### Approach 2
Why: Do we really need the dynamic complexity here? While cool, it's harder to read and just writing out 5 "dumb" validation methods is easier to read and not too tedius. 

How: Just write a separate validation method tied to each field + `isValidx` boolean


### My recommendation
I like approach 2 as it's the most simple and readable approach. But I totally get that approach 1 is 😎

Reply:
Makes sense. I refactored a lot to be like solution two. I added a comment to explain why I have two refs for the validation. The first is just to check if it was checked before. And if it has then we can watch it to still provide the interactivity. But the enum case was removed and the extra function. I went through it and with the comments and organization I think it is much more readable with the same interactivity

Reply:
> after figuring out the clever goodness you're writing here, I think we can simplify this logic in one of two ways.
> 
> ### Approach 1
> Why: A bit simpler, but still keeps your genericized approach
> 
> How: you could pass in the `isValid__` variable directly, then you don't need the `useCorrectValidation` method.
> 
> ### Approach 2
> Why: Do we really need the dynamic complexity here? While cool, it's harder to read and just writing out 5 "dumb" validation methods is easier to read and not too tedius.
> 
> How: Just write a separate validation method tied to each field + `isValidx` boolean
> 
> ### My recommendation
> I like approach 2 as it's the most simple and readable approach. But I totally get that approach 1 is 😎

@theyokohamalife. This is what Phil mentioned when I was using dynamic complexity